app/controllers/system_admin/users_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module SystemAdmin
+  class UsersController < ApplicationController
+    layout "platform_admin"
+    before_action :get_provider
+
+    def new
+      @user = authorize @provider.users.build
+    end
+
+    def create
+      @user = authorize @provider.users.build(user_params)
+      if @user.save
+        redirect_to provider_path(id: params[:provider_id])
```

Comment:
You've already found the provider at this point, so we can pass in the object to rails route helpers:

i.e. `redirect_to provider_path(provider)`

This is more conventional and ensures that we're directing them to a provider that actually exists.